sdk/core/azure-core/src/main/java/com/azure/core/http/rest/StreamResponseBase.java

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.Objects;
+
+/**
+ * REST response with a streaming content and typed headers object.
+ *
+ * @param <HEADERS> The typed headers object.
+ */
+public final class StreamResponseBase<HEADERS> extends ResponseBase<HEADERS, Flux<ByteBuffer>> implements Closeable {
```

Comment:
For now, I'd rather keep them separate as `StreamResponseBase` extend `ResponseBase` whereas `StreamResponse` extends `SimpleResponse` and this would be a flip of how `ResponseBase` implements `Response` and `SimpleResponse` implements `Response`